x-pack/solutions/security/plugins/security_solution/public/detections/components/attacks/table/table_section.tsx

```diff
@@ -0,0 +1,131 @@
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import type { Filter } from '@kbn/es-query';
+import { TableId } from '@kbn/securitysolution-data-table';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { PageScope } from '../../../../data_view_manager/constants';
+import { useGroupTakeActionsItems } from '../../../hooks/alerts_table/use_group_take_action_items';
+import {
+  defaultGroupingOptions,
+  defaultGroupStatsAggregations,
+  defaultGroupStatsRenderer,
+  defaultGroupTitleRenderers,
+} from '../../alerts_table/grouping_settings';
+import { useDataTableFilters } from '../../../../common/hooks/use_data_table_filters';
+import { useDeepEqualSelector } from '../../../../common/hooks/use_selector';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { inputsSelectors } from '../../../../common/store/inputs';
+import { useUserData } from '../../user_info';
+import { useListsConfig } from '../../../containers/detection_engine/lists/use_lists_config';
+import {
+  buildShowBuildingBlockFilter,
+  buildThreatMatchFilter,
+} from '../../alerts_table/default_config';
+import { GroupedAlertsTable } from '../../alerts_table/alerts_grouping';
+import { AlertsTable } from '../../alerts_table';
+
+export const TABLE_SECTION_TEST_ID = 'attacks-page-table-section';
+
+export interface TableSectionProps {
+  /**
+   * DataView used to fetch the alerts data
+   */
+  dataView: DataView;
+}
+
+/**
+ * Renders the alerts table with grouping functionality in the attacks page.
+ */
+export const TableSection = React.memo(({ dataView }: TableSectionProps) => {
+  const getGlobalFiltersQuerySelector = useMemo(
+    () => inputsSelectors.globalFiltersQuerySelector(),
+    []
+  );
+  const globalFilters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
+
+  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+  const query = useDeepEqualSelector(getGlobalQuerySelector);
+
+  const { to, from } = useGlobalTime();
+
+  const [{ loading: userInfoLoading, hasIndexWrite, hasIndexMaintenance }] = useUserData();
+
+  const { loading: listsConfigLoading } = useListsConfig();
+
+  const { showBuildingBlockAlerts, showOnlyThreatIndicatorAlerts } = useDataTableFilters(
+    TableId.alertsOnAlertsPage
+  );
+
+  // AlertsTable manages global filters itself, so not including `filters`
+  const defaultFilters = useMemo(
+    () => [
+      ...buildShowBuildingBlockFilter(showBuildingBlockAlerts),
+      ...buildThreatMatchFilter(showOnlyThreatIndicatorAlerts),
+    ],
+    [showBuildingBlockAlerts, showOnlyThreatIndicatorAlerts]
+  );
+
+  const isLoading = useMemo(
+    () => userInfoLoading || listsConfigLoading,
+    [listsConfigLoading, userInfoLoading]
+  );
+
+  const renderAlertTable = useCallback(
+    (groupingFilters: Filter[]) => {
+      return (
+        <AlertsTable
```

Comment:
I think it's pretty risky to reuse the same alerts table that we have today on the alerts page. That table has a bunch of cell actions, cell formatting and such that might not work for us. I'm thinking for example some of the cell actions could not work for attack documents... And we probably don't want to add more if/else conditions to an already complex codebase...
I would consider doing what I did for EASE and have a new alerts table components, leveraging as many lower level components as possible...

I'm not saying it's the right way to do it, but I think it'd be valuable to look into it (if it has not been done already). Reusing the complex code of the alerts page alerts table scares me a bit...